File: app/src/main/java/id/co/jst/siar/Helpers/sqlite/DBHandlerRAA.java
package id.co.jst.siar.Helpers.sqlite;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.util.Log;
import android.widget.Toast;

import id.co.jst.siar.Models.sqlite.LocationModel;
import id.co.jst.siar.Models.sqlite.RAAModel;

/**
 * Created by endro.ngujiharto on 4/7/2017.
 */

public class DBHandlerRAA extends DBHandlerSQLite {

    public DBHandlerRAA(Context context) {
        super(context);
    }
    private boolean success = false;
    private String message = "";

    // Adding new RAA
    public void addRAA(RAAModel raa) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(KEY_IRPERIODID, raa.getIRPeriodID());
        values.put(KEY_IRASSETNO, raa.getIRAssetNo());
        values.put(KEY_IRMODEL, raa.getIRModel());
        values.put(KEY_IRMFGNO, raa.getIRMFGNo());
        values.put(KEY_IRLOCATIONID, raa.getIRLocationID());
        values.put(KEY_IRGENERATEDATE, raa.getIRGenerateDate());
        values.put(KEY_IRGENERATEUSER, raa.getIRGenerateUser());
        values.put(KEY_IRDEPTCODE, raa.getIRDeptCode());
        // Inserting Row
        db.insert(super.TABLE_RAA, null, values);
        db.close(); // Closing database connection
    }

    // Getting RAA Count
    public int getRAACount() {
        String countQuery = "SELECT * FROM " + super.TABLE_RAA;
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.rawQuery(countQuery, null);

        // return count
        return cursor.getCount();
    }

    // Truncate RAA
    public void truncateRAA() {
        String truncateQuery = "DELETE FROM " + super.TABLE_RAA;
        SQLiteDatabase db = this.getReadableDatabase();
        db.execSQL(truncateQuery);
        db.close();
    }

    // Getting one raa
    public Object[] getRAA(int id) {
        Object[] objects;
        SQLiteDatabase db = this.getReadableDatabase();
        String getRAAQuery = "SELECT " + KEY_IRASSETNO + ", " + KEY_IRMODEL + ", " + KEY_IRMFGNO + ", " + KEY_PLACE + ", " + KEY_ID + " FROM " + TABLE_RAA +
                " INNER JOIN " + TABLE_LOCATION +
                " ON " + KEY_ID + " = " + KEY_IRLOCATIONID +
                " WHERE IRAssetNo = " + id;

        Cursor cursor = db.rawQuery(getRAAQuery, null);

        if (cursor.getCount() > 0){
            cursor.moveToFirst();

            RAAModel raa = new RAAModel();
            LocationModel location = new LocationModel();
            raa.setIRAssetNo(cursor.getInt(0));
            raa.setIRModel(cursor.getString(1));
            raa.setIRMFGNo(cursor.getString(2));
            location.setPl_place(cursor.getString(3));
            objects = new Object[]{raa,location};
            success = true;
        } else {
            message = "Data RAA Tidak Ditemukan.";
            objects = new Object[]{message};
        }

        // return
        return objects;
    }

    // Getting one raa
    public RAAModel getRAAforRAAActual(int id) {
        SQLiteDatabase db = this.getReadableDatabase();
        String getRAAQuery = "SELECT * FROM " + TABLE_RAA +
                " INNER JOIN " + TABLE_LOCATION +
                " ON " + KEY_ID + " = " + KEY_IRLOCATIONID +
                " WHERE IRAssetNo = " + id;

        Cursor cursor = db.rawQuery(getRAAQuery, null);
        if (cursor != null)
            cursor.moveToFirst();

        RAAModel raa = new RAAModel(cursor.getInt(0), cursor.getInt(1), cursor.getString(2), cursor.getString(3), cursor.getInt(4), cursor.getString(5), cursor.getString(6), cursor.getInt(7));

        // return
        return raa;
    }
}

File: app/src/main/java/id/co/jst/siar/Helpers/sqlite/DBHandlerRAA.kt
package id.co.jst.siar.Helpers.sqlite

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteException
import android.util.Log
import android.widget.Toast

import id.co.jst.siar.Models.sqlite.LocationModel
import id.co.jst.siar.Models.sqlite.RAAModel

/**
 * Created by endro.ngujiharto on 4/7/2017.
 */

class DBHandlerRAA(context: Context) : DBHandlerSQLite(context) {
    private var success = false
    private var message = ""

    // Getting RAA Count
    // return count
    val raaCount: Int
        get() {
            val countQuery = "SELECT * FROM " + super.TABLE_RAA
            val db = this.readableDatabase
            val cursor = db.rawQuery(countQuery, null)
            return cursor.count
        }

    // Adding new RAA
    fun addRAA(raa: RAAModel) {
        val db = this.writableDatabase
        val values = ContentValues()
        values.put(DBHandlerSQLite.KEY_IRPERIODID, raa.irPeriodID)
        values.put(DBHandlerSQLite.KEY_IRASSETNO, raa.irAssetNo)
        values.put(DBHandlerSQLite.KEY_IRMODEL, raa.irModel)
        values.put(DBHandlerSQLite.KEY_IRMFGNO, raa.irmfgNo)
        values.put(DBHandlerSQLite.KEY_IRLOCATIONID, raa.irLocationID)
        values.put(DBHandlerSQLite.KEY_IRGENERATEDATE, raa.irGenerateDate)
        values.put(DBHandlerSQLite.KEY_IRGENERATEUSER, raa.irGenerateUser)
        values.put(DBHandlerSQLite.KEY_IRDEPTCODE, raa.irDeptCode)
        // Inserting Row
        db.insert(super.TABLE_RAA, null, values)
        db.close() // Closing database connection
    }

    // Truncate RAA
    fun truncateRAA() {
        val truncateQuery = "DELETE FROM " + super.TABLE_RAA
        val db = this.readableDatabase
        db.execSQL(truncateQuery)
        db.close()
    }

    // Getting one raa
    fun getRAA(id: Int): Array<Any> {
        val objects: Array<Any>
        val db = this.readableDatabase
        val getRAAQuery = "SELECT " + DBHandlerSQLite.KEY_IRASSETNO + ", " + DBHandlerSQLite.KEY_IRMODEL + ", " + DBHandlerSQLite.KEY_IRMFGNO + ", " + DBHandlerSQLite.KEY_PLACE + ", " + DBHandlerSQLite.KEY_ID + " FROM " + DBHandlerSQLite.TABLE_RAA +
                " INNER JOIN " + DBHandlerSQLite.TABLE_LOCATION +
                " ON " + DBHandlerSQLite.KEY_ID + " = " + DBHandlerSQLite.KEY_IRLOCATIONID +
                " WHERE IRAssetNo = " + id

        val cursor = db.rawQuery(getRAAQuery, null)

        if (cursor.count > 0) {
            cursor.moveToFirst()

            val raa = RAAModel()
            val location = LocationModel()
            raa.irAssetNo = cursor.getInt(0)
            raa.irModel = cursor.getString(1)
            raa.irmfgNo = cursor.getString(2)
            location.pl_place = cursor.getString(3)
            objects = arrayOf(raa, location)
            success = true
        } else {
            message = "Data RAA Tidak Ditemukan."
            objects = arrayOf(message)
        }

        // return
        return objects
    }

    // Getting one raa
    fun getRAAforRAAActual(id: Int): RAAModel {
        val db = this.readableDatabase
        val getRAAQuery = "SELECT * FROM " + DBHandlerSQLite.TABLE_RAA +
                " INNER JOIN " + DBHandlerSQLite.TABLE_LOCATION +
                " ON " + DBHandlerSQLite.KEY_ID + " = " + DBHandlerSQLite.KEY_IRLOCATIONID +
                " WHERE IRAssetNo = " + id

        val cursor = db.rawQuery(getRAAQuery, null)
        cursor?.moveToFirst()

// return
        return RAAModel(cursor!!.getInt(0), cursor.getInt(1), cursor.getString(2), cursor.getString(3), cursor.getInt(4), cursor.getString(5), cursor.getString(6), cursor.getInt(7))
    }
}
